mods/default/craftitems.lua

```diff
@@ -84,6 +85,11 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			data = minetest.deserialize(stack:get_metadata())
 		end
 
+		if data and data.owner and data.owner ~= player:get_player_name()  then
```

Comment:
How about using `default.can_interact_with_node()` here? I don't know, it's just a suggestion.